### PR TITLE
bugfix/order patch 

### PR DESCRIPTION
**Pull request by heeler**
This patch is a fix that makes sure the image order is the expected STCZ order. 
Also, there is a new test and test file to make sure that the order is preserved going forward.

### CODE REVIEW

File: c_tests/test_Reader.cpp [hunk 1]
```diff
@@ -541,6 +552,15 @@ TEST_CASE_METHOD(CziCreator5,
   assert(ans[2].h == 256);
 }
 
+TEST_CASE_METHOD(CziCreatorOrder, "test_image_order", "[Reader_image_order]")
+{
+  auto czi = get();
+
+  libCZI::CDimCoordinate dm;
+  auto imCont = czi->readSelected(dm, -1, CORES_FOR_THREADS);
+  assert(true);
```
Comment (toloudis) [inline c_tests/test_Reader.cpp:561]: not testing anything?

File: _aicspylibczi/Reader.cpp [hunk 2]
```diff
@@ -329,6 +334,15 @@ namespace pylibczi {
           throw pylibczi::CdimSelectionZeroImagesException(plane_coord_, m_statistics.dimBounds, "No pyramid0 selectable subblocks.");
       }
       auto charShape = imageFactory.getFixedShape();
+      char *memStart = imageFactory.mem_start();
+      ImageVector &imageVector = imageFactory.images();
+      for( int i = 0; i < imageVector.size(); i++){
+        auto shp = imageVector[i]->coordinatePtr();
+        std::cout << "Offset: "
```
Comment (toloudis) [inline _aicspylibczi/Reader.cpp:341]: can remove this code?

File: _aicspylibczi/Reader.cpp [hunk 3]
```diff
@@ -278,7 +279,11 @@ namespace pylibczi {
       }
       SubblockSortable subblocksToFind(&plane_coord_, index_m_, isMosaic());
       SubblockIndexVec matches = getMatches(subblocksToFind);
-      m_pixelType = matches.front().first.pixelType();
+      int count=0;
+      for(auto x : matches){
+        std::cout << count++ << ":  " << *(x.first.coordinatePtr()) << std::endl;
```
Comment (toloudis) [inline _aicspylibczi/Reader.cpp:284]: can remove this?

File: c_tests/test_Reader.cpp (hunk 1 above)
Comment (heeler) [reply c_tests/test_Reader.cpp]: failed push addressed now

File: _aicspylibczi/Reader.cpp (hunk 2 above)
Comment (heeler) [reply _aicspylibczi/Reader.cpp]: failed push again

**Review by toloudis — APPROVED** [review]
Looks good, the crux is that images were coming out in potentially bad orderings and the switch from vector to set automatically imposes a sorted order.  

Minor concern that we are not really testing enough permutations but assuming this fixes the bug in question.   It would be nice if we could construct CZI containers in the testing code to populate them with known data arrays to test with.

**Comment by heeler** [comment]
I completely agree, the challenge is that I don't know of a way to write a CZI file with synthetic data. Our best long-term solution will be to take an image and heavily manipulate it in Zen and save it out but that requires us to use windows machines that are in the office.

**Comment by JacksonMaxfield** [comment]
I guess I am confused why Seb would say that his image worked when using _just_ `aicspylibczi` and not when using `aicsimageio`. Any idea why this change resolves an upstream?

**Pull request merged**